spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java

```diff
@@ -111,10 +111,9 @@ private HttpClientBuilder lookupHttpClientBuilder(final URL url) {
 			 * which have no placeholders. That is the one we want to use in the case
 			 * there are multiple matches.
 			 */
-			List<String> keys = builderMap.keySet().stream().filter(key -> {
-				String[] tokens = key.split(PLACEHOLDER_PATTERN);
-				return tokens.length == 1;
-			}).collect(Collectors.toList());
+			List<String> keys = builderMap.keySet().stream().filter(
+					key -> !Pattern.compile(PLACEHOLDER_PATTERN).matcher(key).find())
```

Comment:
Could `Pattern.compile()` be moved to a constant?

Reply:
Good call!